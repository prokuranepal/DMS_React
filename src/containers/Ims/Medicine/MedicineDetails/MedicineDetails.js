import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles(() => ({
  modal: {
    position: 'absolute',
    top: '40%',
    left: '40%',
    width: 400,
    background: '#fff',
    borderRadius: '2px',
    boxShadow: '0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23)',
    textAlign: 'center',
  },
  medicineImage: {
    width: '100%',
  },
  editButton: {
    marginBottom: '20px',
  },
}));

/**
* Details of a medicine and can be edited
*
* @param {string} props.medicineName - name of medicine
* @param {string} props.medicineType- type of medicine
* @param {string} props.medicineQuantity - quantity of medicines
* @returns {MedicineDetails} - It returns an image, text and button
*/


const MedicineDetails = (props) => {
  const classes = useStyles();

  return (
    <div className={classes.modal}>
      <img className={classes.medicineImage} src="https://images.unsplash.com/photo-1471864190281-a93a3070b6de?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80" alt="Medicine Image" />
      <div className={classes.medicineDetails}>
        <h3>Medicine Name: {props.medicineName}</h3>
        <h3>Medicine Types: {props.medicineType}</h3>
        <h3>Medicine Quantity: {props.medicineQuantity}</h3>
      </div>
      <Button variant="contained" color="primary" className={classes.editButton}>
        Edit
      </Button>
    </div>
  );
}

export default MedicineDetails;
