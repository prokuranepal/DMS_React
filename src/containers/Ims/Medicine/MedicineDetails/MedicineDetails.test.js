import React from 'react';
import MedicineDetails from './MedicineDetails';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


const setup = (props = {}, state = null) => {
    return (shallow( < MedicineDetails {...props}  />)
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


const dummy_data=
    {
        medicineName:"paracetamol",
        medicineType: "pain killer",
        medicineQuantity: 120
    }


describe('MedicineDetails />', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = setup(dummy_data)
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


it("MedicineDetails snapshot", () => {
        // let wrapper = setup(dummy_data);
        expect(wrapper).toMatchSnapshot();

  
})

it("MedicineDetails Components", () => {

    const h3 = wrapper.find('h3')
    expect(h3.at(0).text()).toEqual("Medicine Name: paracetamol")
    expect(h3.at(1).text()).toEqual("Medicine Types: pain killer")
    expect(h3.at(2).text()).toEqual("Medicine Quantity: 120")


})
})
