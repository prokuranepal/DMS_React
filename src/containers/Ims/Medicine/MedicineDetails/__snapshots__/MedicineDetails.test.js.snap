// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicineDetails /> MedicineDetails snapshot 1`] = `ShallowWrapper {}`;
